File: 2_Kernel/JavaKernel/src/main/java/com/seagosoft/tws/data/OrderStatusData.java
package com.seagosoft.tws.data;

import com.ib.client.Decimal;

public class OrderStatusData {

    public int orderId;
    public String status;
    public Decimal filled;
    public Decimal remaining;
    public double avgFillPrice;
    public int permId;
    public int parentId;
    public double lastFillPrice;
    public int clientId;
    public String whyHeld;
    public double mktCapPrice;

    public OrderStatusData(int orderId, String status,
                           Decimal filled, Decimal remaining, double avgFillPrice,
                           int permId, int parentId, double lastFillPrice, int clientId,
                           String whyHeld, double mktCapPrice) {
        this.orderId = orderId;
        this.status = status;
        this.filled = filled;
        this.remaining = remaining;
        this.avgFillPrice = avgFillPrice;
        this.permId = permId;
        this.parentId = parentId;
        this.lastFillPrice = lastFillPrice;
        this.clientId = clientId;
        this.whyHeld = whyHeld;
        this.mktCapPrice = mktCapPrice;
    }
}

File: 3_OrderSystem/KotlinEWrapperBackend/src/main/kotlin/com/seagosoft/tws/wrapper/structs/OrderStatusData.kt
package com.seagosoft.tws.wrapper.structs

import com.ib.client.Decimal


class OrderStatusData(
    var orderId: Int, var status: String,
    var filled: Decimal, var remaining: Decimal, var avgFillPrice: Double,
    var permId: Int, var parentId: Int, var lastFillPrice: Double, var clientId: Int,
    var whyHeld: String, var mktCapPrice: Double
)